engine/src/main/java/org/terasology/world/chunks/pipeline/tasks/ChunkTaskListenerWrapper.java

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.world.chunks.pipeline.tasks;
+
+import org.joml.Vector3i;
+import org.terasology.world.chunks.Chunk;
+import org.terasology.world.chunks.pipeline.ChunkTask;
+import org.terasology.world.chunks.pipeline.ChunkTaskListener;
+
+/**
+ * Wrapper for {@link ChunkTask} with {@link ChunkTaskListener} feature.
+ * <p>
+ * Fire {@link ChunkTaskListener#onDone(ChunkTask)}  after wrapped {@link ChunkTask}
+ */
+public class ChunkTaskListenerWrapper implements ChunkTask {
+    private final ChunkTask chunkTask;
+    private final ChunkTaskListener listener;
```

Comment:
It is a class named `Wrapper`, I am automatically suspicious. :detective: 

(Another reason for me to take a closer look at the Pipeline class.)

Reply:
Tasks like `Runnable`. This class delegating(wrapping) `ChunkTask`'s run method and append Listener behavior.
Yeah, better to use `Callable` and `Future` but we have TaskMaster :D